relayer/relays/parachain/scanner.go

```diff
@@ -155,112 +144,100 @@ func (s *Scanner) findTasks(
 func (s *Scanner) findTasksImpl(
 	ctx context.Context,
 	lastParaBlockNumber uint64,
-	basicChannelSourceNonces map[types.AccountID]uint64,
+	sourceID uint32,
+	startingNonce uint64,
 ) ([]*Task, error) {
-	basicChannelSourceNonceString := "map["
-	for sourceID, nonce := range basicChannelSourceNonces {
-		basicChannelSourceNonceString += fmt.Sprintf("%v: %v ", hex.EncodeToString(sourceID[:]), nonce)
-	}
-	basicChannelSourceNonceString = strings.Trim(basicChannelSourceNonceString, " ")
-	basicChannelSourceNonceString += "]"
-
 	log.WithFields(log.Fields{
-		"basicSourceNonces": basicChannelSourceNonceString,
-		"latestblockNumber": lastParaBlockNumber,
-	}).Debug("Searching backwards from latest block on parachain to find block with nonces")
-
-	currentBlockNumber := lastParaBlockNumber
-
-	basicChannelScanSources := make(map[types.AccountID]bool, len(basicChannelSourceNonces))
-	for sourceID := range basicChannelSourceNonces {
-		basicChannelScanSources[sourceID] = true
-	}
-	scanBasicChannelDone := len(basicChannelScanSources) == 0
+		"sourceID":          sourceID,
+		"nonce":             startingNonce,
+		"latestBlockNumber": lastParaBlockNumber,
+	}).Debug("Searching backwards from latest block on parachain to find block with nonce")
 
+	scanOutboundQueueDone := false
 	var tasks []*Task
 
-	for !scanBasicChannelDone && currentBlockNumber > 0 {
+	for currentBlockNumber := lastParaBlockNumber; currentBlockNumber > 0; currentBlockNumber-- {
+		if scanOutboundQueueDone {
+			break
+		}
+
 		log.WithFields(log.Fields{
 			"blockNumber": currentBlockNumber,
 		}).Debug("Checking header")
 
 		blockHash, err := s.paraConn.API().RPC.Chain.GetBlockHash(currentBlockNumber)
 		if err != nil {
-			return nil, fmt.Errorf("fetch blockhash for block %v: %w", currentBlockNumber, err)
+			return nil, fmt.Errorf("fetch block hash for block %v: %w", currentBlockNumber, err)
 		}
 
 		header, err := s.paraConn.API().RPC.Chain.GetHeader(blockHash)
 		if err != nil {
-			return nil, fmt.Errorf("fetch header for %v: %w", blockHash.Hex(), err)
+			return nil, fmt.Errorf("fetch header for block hash %v: %w", blockHash.Hex(), err)
 		}
 
 		digestItems, err := ExtractAuxiliaryDigestItems(header.Digest)
 		if err != nil {
 			return nil, err
 		}
-
 		if len(digestItems) == 0 {
 			currentBlockNumber--
```

Comment:
nothing to do with change in this PR, just curious is this line required or a bug should be removed?